nncf/experimental/post_training/compression_builder.py

```diff
@@ -85,17 +105,19 @@ def apply(self, model: ModelType, dataset: Dataset, engine: Engine = None) -> Mo
         backend = infer_backend_from_model(model)
         modified_model = self._get_prepared_model_for_compression(model, backend)
 
+        model_transformer = self._create_model_transformer(model, backend)
+
         if engine is None:
             engine = self._create_engine(backend)
 
         for algorithm in self.algorithms:
+            algorithm.model_transformer = model_transformer
```

Comment:
This is a good question - where should be model_transformer set.
I don't like that we have it in two places: here and in `create_subalgorithms()`.
I suggest to create a new function for that - `_set_model_transformer(algorithms)` and calls it after creating all algorithms.

Reply:
Having such a method as `create_subalgorithms()` isn't a good idea in general 😄 
Can we adapt the property `setter` by setting the needed model transformer for sub-algorithms instead of creating a new method?

Reply:
This is a good point - `create_subalgorithms()` is a bad thing. We should think how to avoid it.

Reply:
I created a new comment on that